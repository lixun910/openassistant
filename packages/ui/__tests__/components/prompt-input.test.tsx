// SPDX-License-Identifier: MIT
// Copyright contributors to the openassistant project

/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom';
import { render, act } from '@testing-library/react';
import React from 'react';
import PromptInput from '../../src/components/prompt-input';

test('renders correctly', async () => {
  let container;
  await act(async () => {
    const result = render(<PromptInput />);
    container = result.container;
  });
  expect(container).toMatchSnapshot();
});

test('renders correctly with placeholder', () => {
  const { container } = render(<PromptInput placeholder="Say something" />);

  // Check if the placeholder is rendered
  const input = container.querySelector('textarea');
  expect(input).toHaveAttribute('placeholder', 'Say something');
});

test('renders correctly with value', () => {
  const { container } = render(<PromptInput value="Hello, world!" />);

  // Check if the value is rendered
  const input = container.querySelector('textarea');
  expect(input).toHaveValue('Hello, world!');
});

test('render with end content', () => {
  const { container } = render(<PromptInput endContent={<div>End</div>} />);
  const endContent = container.querySelector('div');
  expect(endContent).toBeInTheDocument();

  // the content of end content should be "End"
  expect(endContent).toHaveTextContent('End');
});

test('render with classNames', () => {
  const { container } = render(<PromptInput classNames={{ input: 'bg-red-500' }} />);
  const input = container.querySelector('textarea');
  expect(input).toHaveClass('bg-red-500');
});

test('render with classNames label will be hidden', () => {
  const { container } = render(<PromptInput classNames={{ label: 'bg-red-500' }} />);
  const label = container.querySelector('textarea');
  expect(label).not.toHaveClass('bg-red-500');
});
