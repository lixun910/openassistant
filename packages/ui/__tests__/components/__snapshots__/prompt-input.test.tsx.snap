// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="group flex flex-col data-[hidden=true]:hidden w-full min-h-[40px]"
    data-filled="true"
    data-filled-within="true"
    data-slot="base"
  >
    <div
      class="relative w-full inline-flex tap-highlight-transparent flex-row items-center shadow-sm px-3 gap-3 border-medium border-default-200 data-[hover=true]:border-default-400 group-data-[focus=true]:border-default-foreground h-10 min-h-10 rounded-large !h-auto transition-background !duration-150 transition-colors motion-reduce:transition-none py-2"
      data-slot="input-wrapper"
      style="cursor: text;"
    >
      <div
        class="inline-flex w-full h-full box-border items-start group-data-[has-label=true]:items-start"
        data-slot="inner-wrapper"
      >
        <textarea
          aria-label="Prompt"
          class="w-full font-normal bg-transparent !outline-none placeholder:text-foreground-500 focus-visible:outline-none data-[has-start-content=true]:ps-1.5 data-[has-end-content=true]:pe-1.5 file:cursor-pointer file:bg-transparent file:border-0 autofill:bg-transparent bg-clip-text text-small resize-none data-[hide-scroll=true]:scrollbar-hide transition-height !duration-100 motion-reduce:transition-none py-0 pe-0"
          data-hide-scroll="true"
          data-slot="input"
          id="react-aria-:r0:"
          placeholder="Enter a prompt here"
          title=""
        />
      </div>
    </div>
  </div>
</div>
`;
